src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -492,7 +481,20 @@ class CVulkanLogicalDevice final : public ILogicalDevice
 
             const IGPUDescriptorSetLayout* layout = pDescriptorWrites[i].dstSet->getLayout();
             if (layout->getAPIType() != EAT_VULKAN)
+            {
+                auto logger = (m_physicalDevice->getDebugCallback()) ? m_physicalDevice->getDebugCallback()->getLogger() : nullptr;
+                if (logger)
+                {
+                    const char* dsDebugName = pDescriptorWrites[i].dstSet->getDebugName();
+                    const char* dsLayoutDebugName = pDescriptorWrites[i].dstSet->getDebugName();
+                    if (dsDebugName && dsLayoutDebugName)
+                        logger->log("Descriptor set layout (%s, %p) of the descriptor set (%s, %p) uses a different backend than the logical device it was created with. Skipping the descriptor set..", system::ILogger::ELL_ERROR, dsLayoutDebugName, layout, dsDebugName, pDescriptorWrites[i].dstSet);
+                    else
+                        logger->log("Descriptor set layout (%p) of the descriptor set (%p) uses a different backend than the logical device it was created with. Skipping the descriptor set..", system::ILogger::ELL_ERROR, layout, pDescriptorWrites[i].dstSet);
+
+                }
                 continue;
+            }
```

Comment:
you can joist device compatibility checks to `ILogicalDevice::updateDescriptorSets`

Reply:
also if any of the `dstSet` or `srcSet` fail this, you can fail the entire call and just not proceed.

Reply:
still not resolved

Reply:
As discussed on discord, all the validation + the `_impl` call needs to happen before `ds->processWrites` and `ds->processCopies`

Reply:
I think the `updateDescriptorSets_impl` does not fail easily (the corresponding Vulkan call, `updateDescriptorSets_impl` doesn't even have a return value) so we can let it happen after `processWrites` and `processCopies`.

Reply:
Resolved.